src/nucleus/tt.ml

```diff
@@ -661,7 +662,7 @@ and print_structure ?max_level ~penv (s,shares) es ppf =
   let rec fold acc es ls shares = match ls, shares with
     | [], [] -> List.rev acc
     | _::ls, (Inr _) :: shares -> fold acc es ls shares
-    | l::ls, (Inl _)::shares ->
+    | l::ls, (Inl _) :: shares ->
```

Comment:
This isn't the right place to comment this, but can you please rename the `sum` type and the `Inr` and `Inl` to something more descriptive? It's only used for sharing, so the `Inl` and `Inr` should be called something else that described their role.